### PR TITLE
Add script for Maven Central publishing

### PR DESCRIPTION
**Pull request by Eskibear**
Usage: 
* create staging repo, sign and upload artifacts.
`node publishMaven.js -task upload`

* promote staging repo to Central
`node publishMaven.js -task promote`

### CODE REVIEW

File: scripts/publishMaven.js [hunk 1]
```diff
@@ -0,0 +1,270 @@
+/**
+ * Usage:
+ * node script.js -task [upload|promote]
```
Comment (testforstephen) [inline scripts/publishMaven.js]: node publishMaven.js - task ...

File: scripts/publishMaven.js [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+            return true;
+        }
+        // use system sleep command to pause the program.
+        childProcess.execSync(`sleep 6s`);
+    }
+    return false;
+}
+
+function checkPrerequisite(configs) {
+    if (!configs.releaseVersion) {
```
Comment (testforstephen) [inline scripts/publishMaven.js]: Null check for other necessary keys either?

File: scripts/publishMaven.js [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+        console.error("\n\n[Failure] Promoting staging repository failed.");
+        console.error(!message ? ex : message.toString());
+        process.exit(1);
+    }
+    const success = isReleased(configs);
+    console.log("Below is the public repository url, you could manually validate it.");
+    console.log(`https://oss.sonatype.org/content/groups/public/${configs.groupId.replace(/\./g, "/")}`);
+    console.log("\n\n");
+    if (success) {
+        console.log("\n\n[Success] Nexus: Promote completion.");
```
Comment (testforstephen) [inline scripts/publishMaven.js]: Promote succeeded

File: scripts/publishMaven.js (hunk 1 above)
Comment (Eskibear) [reply scripts/publishMaven.js]: sure

File: scripts/publishMaven.js (hunk 2 above)
Comment (Eskibear) [reply scripts/publishMaven.js]: will add artifactFolder and user/pass etc.

File: scripts/publishMaven.js (hunk 3 above)
Comment (Eskibear) [reply scripts/publishMaven.js]: ok

**Comment by Eskibear** [comment]
> Considering adding license header?

Not neccessarily I think. As there's no header in `pom.xml` (more important), why bother adding it to this script?

**Pull request merged**